### ISSUE TITLE
Oembed empty non empty condition

### ISSUE THREAD

**Issue by fastmarketo** [CMB2/CMB2#1140]
Check If Oembed is empty or not empty.

I am using this kind of situation →
![show_video_show_image](https://user-images.githubusercontent.com/6827222/40634470-d8a60ad0-6312-11e8-98cf-8f080596a9ff.png)

Buy when show video is not selected the loop doesn't work. so I want to check if the Oembed input box is empty or not what condition should I use?

**Comment by fastmarketo** [comment]
Yes, based on the meta I have to display something on the front end.
I short how to check the condition that oembed (meta information) is filled or not filled?

**Comment by tw2113** [comment]
what code are you using now for the spot to be displayed in? Something like an `if ( ! empty( $value ) ) { ... }` should help. However, need to make sure you're not checking for a value of like a URL that should be embedded, vs the result of trying to embed. You would likely always have a URL to embed, but that doesn't mean it returned successful content.

**Comment by fastmarketo** [comment]
It is WP meta → It is used in a Loop →
https://www.screencast.com/t/mWBSQNge3rW

The issue is although the post has Video as O-Embed, in the drop-down if it is not selected to be shown on the front end, instead, an image is chosen to be shown than although the Oembed has  Video URL, it treats as if it has no video and the post is skipped.

Please seee the image attached in the first post.

**Comment by tw2113** [comment]
From what I can see, you have 2 different fields, and you're only checking for the value of one of them. You need to do something with the "oEmbed" post meta you've set up, likely a URL, depending on the value of this `$image_or_video`.

Perhaps the `[embed]` shortcode briefly demo'd at the top of https://codex.wordpress.org/Embeds ?

I think also applying the `the_content` filter to the meta value would work as well, but it potentially could add other bits from the site that may not be desired.

Last Idea I have would be https://codex.wordpress.org/Function_Reference/wp_oembed_get

Nonetheless, you have part of things done, but not everything needed, given 2 different meta fields for this setup.

**Comment by fastmarketo** [comment]
Suppose I have this →
`<?php $url = esc_url( get_post_meta( get_the_ID(), 'video_oembed', true ) ); ?>`

Now how should I check if the $url is empty or it has a valid video URL? Some PHP stuff right. Please help me to get this done.

(It will be empty if no video URL has been fed in the post meta).

**Comment by tw2113** [comment]
No, not quite what you're wanting, if I've followed correctly,  you want to act depending on if the url provides a preview or oembed data right? That comes after checking if you have a URL at all.

So it'd be more a case of this:

```
<?php
$embed_code = wp_oembed_get( $url );
if ( false === $embed_code ) {
    // Do something else here
} else {
    // We received data back, so we have something to work with.
    echo $embed_code;
}
```
